src/app/pages/common/entity/table/table.component.ts

```diff
@@ -21,10 +21,30 @@ export interface AppTableHeaderAction {
 
 export interface AppTableHeaderExtraAction extends AppTableHeaderAction {}
 
+export interface AppTableColumn {
+  name?: string;
+  name1?: string;
+  name2?: string;
+  prop?: string;
+  prop1?: string;
+  prop2?: string;
+  checkbox?: boolean;
+  onChange?(data: any): void;
+  width?: string;
+  state?: any;
+  button?: boolean;
+  tooltip?: string;
+  iconTooltip?: string;
+  enableMatTooltip?: boolean;
+  hidden?: boolean;
+  listview?: boolean;
+  getIcon?(element: any, prop: string): void;
+}
+
 export interface AppTableConfig<P = any> {
   title?: string;
   titleHref?: string;
-  columns: any[];
+  columns: AppTableColumn[];
```

Comment:
The could just be called `TableColumn`

Reply:
It may be confusing when it is associated with `TableComponent` or `EntityTableComponent`. The other interfaces in this file also have the prefix.